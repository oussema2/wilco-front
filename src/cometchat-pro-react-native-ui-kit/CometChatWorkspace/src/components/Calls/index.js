export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
