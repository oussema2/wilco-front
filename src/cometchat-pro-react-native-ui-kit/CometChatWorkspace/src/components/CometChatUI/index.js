/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
