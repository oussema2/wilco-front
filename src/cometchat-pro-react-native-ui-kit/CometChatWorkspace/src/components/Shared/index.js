export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
