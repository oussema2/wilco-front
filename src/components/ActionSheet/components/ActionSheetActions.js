import React from 'react';
import { observer } from 'mobx-react';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { ActionSheetGradient } from '../../ActionSheetGradient';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetActions = () => {
	const actionSheetService = useActionSheetService();

	return (
		<ActionSheetGradient style={styles.actions}>
			{actionSheetService.actions.map( ( action ) => (
				<ActionSheetButton
					key={action.title}
					{...action}
					onPress={() => {
						actionSheetService.close();
						action.onPress();
					}}
				/>
			) )}
		</ActionSheetGradient>
	);
};

export default observer( ActionSheetActions );
