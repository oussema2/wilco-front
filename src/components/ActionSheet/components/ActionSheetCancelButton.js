import React from 'react';
import { observer } from 'mobx-react';

import { View } from 'react-native';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetCancelButton = () => {
	const actionSheetService = useActionSheetService();

	return (
		<View style={styles.cancelButtonContainer}>
			<ActionSheetButton
				title="Cancel"
				onPress={() => actionSheetService.close()}
			/>
		</View>
	);
};

export default observer( ActionSheetCancelButton );
