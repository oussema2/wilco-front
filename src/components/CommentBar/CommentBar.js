import React from 'react';
import { View } from 'react-native';
import PropTypes from 'prop-types';
import noop from '../../helpers/noop';
import { TextButton } from '../TextButton';
import { styles } from './CommentBar.styles';
import { MultilineInput } from '../MultilineInput';

const CommentBar = ( {
	testID, value, onChange, onReplyPressed
} ) => (
	<View testID={testID} style={styles.containerView}>
		<MultilineInput testID="comment-Input" capitalizeFirstLetter value={value} onChange={onChange} />
		<View style={styles.replyButton}>
			<TextButton testID="reply-TextButton" title="Reply" onPress={onReplyPressed} />
		</View>
	</View>
);

CommentBar.propTypes = {
	testID: PropTypes.string,
	value: PropTypes.string,
	onChange: PropTypes.func,
	onReplyPressed: PropTypes.func
};

CommentBar.defaultProps = {
	testID: 'commentBar-Component',
	value: '',
	onChange: noop,
	onReplyPressed: noop
};

export default CommentBar;
