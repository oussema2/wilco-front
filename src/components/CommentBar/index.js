export { default as CommentBar } from './CommentBar';
