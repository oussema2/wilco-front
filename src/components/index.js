export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
