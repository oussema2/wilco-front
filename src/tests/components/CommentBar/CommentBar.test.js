import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { CommentBar } from '../../../components/CommentBar';

describe( '<CommentBar />', () => {
	let component;
	const testID = 'testing-CommentBar-Component';
	const onReplyPressed = jest.fn();

	const setUp = () => {
		component = render( <CommentBar testID={testID} onReplyPressed={onReplyPressed} /> );
	};

	beforeEach( () => {
		setUp();
	} );

	describe( 'with the default props', () => {
		it( 'renders the CommentBar component correctly', () => {
			expect( component.queryByTestId( testID ) ).toBeDefined();
			expect( component.queryByTestId( 'comment-Input' ) ).toBeDefined();
			expect( component.queryByTestId( 'reply-TextButton' ) ).toBeDefined();

			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'when reply button is pressed', () => {
		it( 'calls onPress', () => {
			const replyButton = component.queryByTestId( 'reply-TextButton' );
			fireEvent.press( replyButton );
			expect( onReplyPressed ).toHaveBeenCalledTimes( 1 );
		} );
	} );
} );
