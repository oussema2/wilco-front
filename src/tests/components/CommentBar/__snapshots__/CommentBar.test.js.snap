// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentBar /> with the default props renders the CommentBar component correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginVertical": 8,
    }
  }
  testID="testing-CommentBar-Component"
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <TextInput
      autoCapitalize="sentences"
      editable={true}
      multiline={true}
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=""
      placeholderTextColor="#A0A0A0"
      scrollEnabled={false}
      selectionColor="#12A3BE"
      style={
        Object {
          "backgroundColor": "#F2F2F2",
          "borderRadius": 8,
          "color": "#000000",
          "fontFamily": "NotoSansJP-Bold",
          "fontSize": 14,
          "fontWeight": "normal",
          "letterSpacing": 0,
          "lineHeight": 20,
          "minHeight": 48,
          "paddingBottom": 16,
          "paddingHorizontal": 12,
          "paddingTop": 16,
          "textAlignVertical": "top",
        }
      }
      testID="comment-Input"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "marginLeft": 12,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 3,
          },
          undefined,
          false,
        ]
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "busy": false,
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="reply-TextButton"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "flex-end",
              "backgroundColor": "transparent",
              "borderColor": "#2089dc",
              "borderRadius": 26,
              "borderWidth": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#12A3BE",
                "fontFamily": "NotoSansJP-Bold",
                "fontSize": 15,
                "fontWeight": undefined,
                "letterSpacing": 0,
                "lineHeight": 20,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Reply
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
